lib/events.js

```diff
@@ -240,13 +240,11 @@ function identicalSequenceRange(a, b) {
 
 function enhanceStackTrace(err, own) {
   let ctorInfo = '';
-  try {
-    const { name } = this.constructor;
-    if (name !== 'EventEmitter')
-      ctorInfo = ` on ${name} instance`;
-  } catch {}
-  const sep = `\nEmitted 'error' event${ctorInfo} at:\n`;
+  const { name } = this.constructor || {};
+  if (name && name !== 'EventEmitter')
+    ctorInfo = ` on ${name} instance`;
```

Comment:
I appreciate the intent but the new code isn't equivalent to the old code. The try/catch guards against "poisoned" properties. To illustrate:
```js
const that = {
  get constructor() { throw 'fail' }
}

// okay
try {
  const { name } = that.constructor
} catch {}

// not okay, throws
const { name } = that.constructor
```

Reply:
Yes, you are right.
I there any test that cover this case?

Reply:
If `make test` passes with your change, the answer is 'no'. :-)

A regression test would be welcome, of course!

Reply:
"poisoned" properties are true for almost every line in the node/lib code.
But you don't see try-catch on every property/function access.

Reply:
That's true but this function deals with stack traces. You don't want an exception/stack trace from a poisoned property hiding the real exception/stack trace.